web/files/files.go

```diff
@@ -714,13 +716,72 @@ func DestroyFileHandler(c echo.Context) error {
 	return c.NoContent(204)
 }
 
+const maxMangoLimit = 100
+
+// FindFilesMango is the route POST /files/_find
+// used to retrieve files and their metadata from a mango query.
+func FindFilesMango(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+	var findRequest map[string]interface{}
+
+	if err := c.Bind(&findRequest); err != nil {
+		return jsonapi.NewError(http.StatusBadRequest, err)
+	}
+
+	if err := permissions.AllowWholeType(c, permissions.GET, consts.Files); err != nil {
+		return err
+	}
+
+	limit, hasLimit := findRequest["limit"].(float64)
+	if !hasLimit || limit > maxMangoLimit {
+		limit = 100
+	}
+	skip := 0
+	skipF64, hasSkip := findRequest["skip"].(float64)
+	if hasSkip {
+		skip = int(skipF64)
+	}
+
+	// add 1 so we know if there is more.
+	findRequest["limit"] = limit + 1
+
+	var results []vfs.DirOrFileDoc
+	err := couchdb.FindDocsRaw(instance, consts.Files, &findRequest, &results)
+	if err != nil {
+		return err
+	}
+
+	var total int
+	if len(results) > int(limit) {
+		total = math.MaxInt32 - 1          // we dont know the actual number
```

Comment:
It's feasible, but a bit complicated: you have to parse the mango query selector and add a $gt condition on last sorting field.